### PR TITLE
feat: allow ➡️ to add chat context in background

### PR DESCRIPTION
**Pull request by joyceerhl**
cc @tylerleonhardt

### CODE REVIEW

File: src/vs/workbench/contrib/chat/browser/actions/chatContextActions.ts [hunk 1]
```diff
@@ -56,10 +56,26 @@ class AttachContextAction extends Action2 {
...
+			id: 'id' in p && typeof p.id === 'string' ? p.id :
+				'resource' in p && URI.isUri(p.resource) ? `${SelectAndInsertFileAction.Name}:${p.resource.toString()}` : ''
+		})));
+	}
+
 	override async run(accessor: ServicesAccessor, ...args: any[]): Promise<void> {
 		const quickInputService = accessor.get(IQuickInputService);
 		const chatVariablesService = accessor.get(IChatVariablesService);
 		const widgetService = accessor.get(IChatWidgetService);
+		const context: { widget?: IChatWidget } | undefined = args[0];
```
Comment (meganrogge) [inline src/vs/workbench/contrib/chat/browser/actions/chatContextActions.ts:74]: could `args` be undefined here, thus this would throw?

**Review by meganrogge — APPROVED** [review]

File: src/vs/platform/quickinput/common/quickAccess.ts [hunk 2]
```diff
@@ -54,7 +60,7 @@ export interface IQuickAccessOptions {
 	/**
 	 * Provider specific options for this particular showing of the
 	 * quick access.
-	 */
+	*/
```
Comment (roblourens) [inline src/vs/platform/quickinput/common/quickAccess.ts:63]: These got misaligned

File: src/vs/platform/quickinput/browser/pickerQuickAccess.ts [hunk 3]
```diff
@@ -326,6 +326,14 @@ export abstract class PickerQuickAccessProvider<T extends IPickerQuickAccessItem
 
 		// Accept the pick on accept and hide picker
 		disposables.add(picker.onDidAccept(event => {
+			if (runOptions?.handleAccept) {
```
Comment (roblourens) [inline src/vs/platform/quickinput/browser/pickerQuickAccess.ts:329]: Can't you use the 'accept' callback on the item to do the same thing? I don't follow

Comment (joyceerhl) [reply src/vs/platform/quickinput/browser/pickerQuickAccess.ts:329]: We don't control the items in the picker, they are populated from the anything quick access in this case

Comment (joyceerhl) [reply src/vs/platform/quickinput/browser/pickerQuickAccess.ts:329]: We could overwrite the `accept` handler for all items in the anything quick access layer, but this feels more explicit

**Pull request merged**